google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ModelTableDefinition.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.api.services.bigquery.model.Table;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * A Google BigQuery Model table definition. This definition is used to represent a BigQuery
+ * ML model.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-create#models_in_bqml_name">BigQuery ML Model</a>
+ */
+@AutoValue
+public abstract class ModelTableDefinition extends TableDefinition {
+
+  //TODO(gsabhnani): Update this id??
```

Comment:
I'm not sure where the original ID came from. I don't think it matters so long as it's consistent and we update it when we make a breaking change to the class.

Reply:
I copied this from StandardTableDefinition.java file. Seems like each class has a unique id and I am not sure if we need another one for this.